pkgs/misc/arm-trusted-firmware/default.nix

```diff
@@ -28,12 +28,20 @@ let
     pname = "arm-trusted-firmware${lib.optionalString (platform != null) "-${platform}"}";
     version = "2.10.0";
 
-    src = fetchFromGitHub {
-      owner = "ARM-software";
-      repo = "arm-trusted-firmware";
-      rev = "v${version}";
-      hash = "sha256-CAuftVST9Fje/DWaaoX0K2SfWwlGMaUFG4huuwsTOSU=";
-    };
+    # TODO: remove this condition once mainline ATF has support for the rk3588 platform
+    src = if platform == "rk3588" then (
+      fetchurl {
+        url = "https://gitlab.collabora.com/hardware-enablement/rockchip-3588/trusted-firmware-a/-/archive/rk3588/trusted-firmware-a-rk3588.tar.gz?002d8e85ce5f4f06ebc2c2c52b4923a514bfa701";
+        hash = "sha256-hw/LW5DaXfvlpIpFTMp1T/yok9K4XJyS2btL+BII/UA=";
+      }
```

Comment:
- Use `fetchFromGitLab`, which can also work with other instances than gitlab dot com.
 - Remove the `if` condition and pass `src` down where you define `armTrustedFirmwareRK3588`

The extra args given to `buildArmTrustedFirmware` allows overriding the default args.

Reply:
Understood, will tend to this tomorrow morning :)

Reply:
On the topic of creating a separate PR, is it okay to introduce the `rkbin` package in this PR or should that be separate too?

Reply:
Does https://github.com/NixOS/nixpkgs/pull/292667/commits/c86043fd665093bb9a11fa52aef410536272289f look like what you asked?